lib/_http_client.js

```diff
@@ -569,10 +569,20 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
 };
 
 ClientRequest.prototype.setNoDelay = function() {
-  this._deferToConnect('setNoDelay', arguments);
+  var argsLength = arguments.length;
+  var args = new Array(argsLength);
+  for (var i = 0; i < argsLength; i++) {
+    args[i] = arguments[i];
+  }
```

Comment:
While you're at it, I suggest just optimizing as much as possible:

https://gist.github.com/Fishrock123/98c35a0c745cb59d7496